src/types/User.ts

```diff
@@ -49,3 +67,83 @@ export enum UsernameVisibility {
   hover = "hover",
   inline = "inline",
 }
+
+export const ExperienceSchema: Yup.ObjectSchema<Experience> = Yup.object()
+  .shape({
+    // @debt refactor bartender to be potentially undefined. Or can we remove it entirely?
+    bartender: Yup.lazy(() => UserSchema.required()),
+    table: Yup.string().nullable(),
+    row: Yup.number().nullable(),
+    column: Yup.number().nullable(),
+  })
+  .required();
+
+export const UserExperienceDataSchema = Yup.lazy<
+  UserExperienceData | undefined
+>((data) => {
+  const lazyObjectShape = Object.fromEntries(
+    Object.keys(data ?? {}).map((key) => [key, ExperienceSchema])
+  );
+
+  return Yup.object().shape(lazyObjectShape).noUnknown();
+});
+
+export const VideoStateSchema: Yup.ObjectSchema<VideoState> = Yup.object()
+  .shape({
+    inRoomOwnedBy: Yup.string(),
+    removedParticipantUids: Yup.array().of(Yup.string().required()),
+  })
+  .required();
+
+export const MyPersonalizedScheduleSchema = Yup.lazy<
+  MyPersonalizedSchedule | undefined
+>((data) => {
+  const lazyObjectShape = Object.fromEntries(
+    Object.keys(data ?? {}).map((key) => [
+      key,
+      Yup.array().of(Yup.string().required()),
+    ])
+  );
+
+  return Yup.object().shape(lazyObjectShape).noUnknown();
+});
+
+/**
+ * User validation schema.
+ *
+ * @see User
+ */
+export const UserSchema: Yup.ObjectSchema<User> = Yup.object()
+  .shape<User>({
+    partyName: Yup.string(),
+    pictureUrl: Yup.string(),
+    anonMode: Yup.boolean(),
+    mirrorVideo: Yup.boolean(),
+    status: Yup.string(),
+    data: UserExperienceDataSchema,
+    myPersonalizedSchedule: MyPersonalizedScheduleSchema,
+    enteredVenueIds: Yup.array().of(Yup.string().required()),
+
+    // Legacy?
+    room: Yup.string(),
+    video: VideoStateSchema.notRequired(),
+    kidsMode: Yup.boolean(),
```

Comment:
These are legacy AF. Can you please add tech debt?

Reply:
That's what the `// Legacy?` was for, but i'll make them more explicitly `@debt` and probably even `@deprecated` on the main object.

Reply:
fixed in b3440d05433547c7516d6113f5446f2cc10a6b3f